src/toil/provisioners/abstractProvisioner.py

```diff
@@ -402,6 +402,10 @@ def _getCloudConfigUserData(self, role, masterPublicKey=None, keyPath=None, pree
         if keyPath:
             mesosArgs = keyPath + ' ' + mesosArgs
             entryPoint = "waitForKey.sh"
+        customDockerInitCommand = customDockerInitCmd()
+        if customDockerInitCommand:
+            mesosArgs = " ".join(["'" + customDockerInitCommand + "'", entryPoint, mesosArgs])
+            entryPoint = "customDockerInit.sh"
```

Comment:
Oh, this is chaining on top of `waitForKey.sh` if it is used. It looked to me like it was replacing it.

Reply:
Correct. We could actually do the chaining in-place here and remove the need for `waitForKey.sh` altogether (I can do that in another PR if desired).